src/components/DateInput.jsx

```diff
@@ -106,7 +104,7 @@ export default class DateInput extends React.Component {
           type="text"
           id={id}
           name={id}
-          ref={(ref) => { this.inputRef = ref; }}
+          ref={(ref) => { this.input = ref; }}
```

Comment:
using arrows as ref callbacks (instead of bound instance methods) means it can sometimes get null. I think we should use a bound instance method instead, throughout.